### PR TITLE
Hrt 53 create print card page v2

### PR DESCRIPTION
**Pull request by TCHE614**
### Purpose
Creating the page to allow the user to print their cards. 

This PR still uses dummy data but the backend ports are set up.

[](https://softeng761team5.atlassian.net/browse/HRT-27)

### Approach
Created a new page and component to design the dimensions for printing purposes

### Does this PR introduce a breaking change?

- [x] Yes
- [ ] No

Added react-to-print so you will have to use ```npm install --save react-to-print```
#### Does this PR involve a UI change?

- [x] Yes
- [ ] No

#### Screenshot(s) of UI after this PR
image
image

### CODE REVIEW

**Review by gargi-bansal — CHANGES_REQUESTED** [review]
Looks good for the most part, the print function is looking really neat! Just a few things I've noticed.

- For some reason, "Statement 1.", "Statement 2." aren't centered. Is there a reason for this?
- The dimension numbers should probably start from 1, not 0. Starting from 0 won't really make sense in this context. Also, can they be a bit more further away from the top-left corner? Tbh I think the cards on the left-hand side need a bit more padding in general. The padding on cards on the right-hand side is good!
- Is there any way the more important information could be highlighted (i.e via bolding or italicising the important text). The cards on her PowerPoint seem to use bolding somewhat effectively. At the moment there is just a lot of text with the same styling, so some emphasis would help greatly.
- I remember Claire mentioning that the "Type" of the dimension was also very important. The two types are Belief and Practice dimensions (also shown in her PowerPoint). The type is also saved in the dummy data, so you'll have to figure out where to place it. You may have to experiment with font sizes and margins or see if you can make the print page wider/larger. She also associates two different colours with the types of dimensions (blue and orange in her PowerPoint), so it might be useful to colour code the cards as well. She mentioned the colours could change as long as they were distinct. 
- If you wanted to, you could very easily pinpoint the point on the continuum/slider that the card corresponds too (we've already done this in the DisplayCards and Preview page for reference, except it's easier in your case since the value will never change). I'm assuming you're already taking the dummy data which contains this info as well. This would be a nice visual aid rather than just having to rely on reading to figure out where on the spectrum you lie.
- A final little nitpick. Is it possible to round the corners of the cards? I feel like a border-radius of 10 or 20px would be nice and a bit more consistent with the rest of the UI. At the moment it's a bit sharp and jarring. This isn't a major though.

Sorry for the super long review, I just get super meticulous with UI 😂 Do let me know if you need any help!

File: src/components/PrintDimension.tsx [hunk 1]
```diff
@@ -0,0 +1,55 @@
+import { Slider, Switch } from "antd";
```
Comment (gargi-bansal) [inline src/components/PrintDimension.tsx]: Make sure to remove any unused import statements, there are a few on this page :)

File: src/components/PrintDimension.tsx [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    })
+      .then((res) => res.json())
+      .then((res) => {
+        //Get data and set it to Dimensions
+      })
+      .catch((e) => console.log(e));
+  }, []);
+
+  const allDimensions = charts[0].dimensions;
+  const something = allDimensions[props.dimensionVallue];
```
Comment (gargi-bansal) [inline src/components/PrintDimension.tsx]: Probably need a better variable name than "something" haha. Also, value is spelt with a double L, probably best to fix this.

File: src/components/PrintDimension.tsx [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    <>
+      <span className="printDimensionText">Pedagogical Dimension:</span>
+      <span className="printTitleText">
+        {allDimensions[props.dimensionVallue].name}
+      </span>
+      <span className="printContinuumText">Continuum:</span>
+      <Slider
+        className="printSlider"
+        range
+        defaultValue={[0, 100]}
```
Comment (gargi-bansal) [inline src/components/PrintDimension.tsx]: You could easily put the slider position here (as mentioned in my review). It just takes another parameter called "value" which is a number between 0 and 100.

File: src/pages/PrintCards.tsx [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+import { Button } from "antd";
+import Header from "../components/Header";
+import Dimension from "../components/PrintDimension";
+import "../styles/PrintCards.css";
+import "../styles/Footer.css";
+import { API_DOMAIN } from "../config";
+
+import charts from "../dummyData/charts";
+
+const chartID = "";
```
Comment (gargi-bansal) [inline src/pages/PrintCards.tsx]: Is this constant necessary? It's only used in line 67 but doesn't do anything (yet at least)

File: package-lock.json [hunk 5]
```diff
@@ -0,0 +1,19923 @@
+{
```
Comment (gargi-bansal) [inline package-lock.json]: I don't think this file is supposed to be committed? I could be wrong @PreetPatel

**Review by mitchfaulcon — CHANGES_REQUESTED** [review]
I notice that both cards in a dimension have the same "Statement reflects 'XYZ' purpose" (I think it's using the 100 mark value every time?) so these should be updated to match what the card really corresponds to.

The URL also stays as /PrintCards# instead of going back to /PrintCards when printing is cancelled but I'm not sure whether that needs to be fixed or if it's fine.

Other than Gargi's changes, everything else looks good :)

File: package-lock.json (hunk 5 above)
Comment (preetpatel) [reply package-lock.json]: yup this shouldn't be here.. never use NPM :)))

File: src/components/PrintDimension.tsx (hunk 1 above)
Comment (TCHE614) [reply src/components/PrintDimension.tsx]: Yep good catch :)

File: src/components/PrintDimension.tsx (hunk 2 above)
Comment (TCHE614) [reply src/components/PrintDimension.tsx]: Ahaha forgot to change that

File: src/components/PrintDimension.tsx (hunk 3 above)
Comment (TCHE614) [reply src/components/PrintDimension.tsx]: Yup sounds good

File: src/pages/PrintCards.tsx (hunk 4 above)
Comment (TCHE614) [reply src/pages/PrintCards.tsx]: Yeah i don't think its need in this right now

File: package-lock.json (hunk 5 above)
Comment (TCHE614) [reply package-lock.json]: Yeah, I'm not sure if I should add package.json or package-lock.json for react-to-print. 

edit: didn't see Preets message will remove the file now

**Comment by TCHE614** [comment]
Hi @gargi-bansal and @mitchfaulcon, I've addressed all the changes from your feedback Please have a look over again thanks :)

**Review by mitchfaulcon — DISMISSED** [review]

**Review by gargi-bansal — DISMISSED** [review]
@TCHE614 The new changes look amazing! Last tiny thing, could you make the background colour of the right-hand side cards white before you merge the branch in?

**Review by gargi-bansal — APPROVED** [review]
Looks so good, thanks for making the changes!

**Review by mitchfaulcon — APPROVED** [review]
Thanks for all the changes, looks great!

**Pull request merged**